File: November/Day10/1047._Remove_All_Adjacent_Duplicates_In_String.java
class Solution {
    public String removeDuplicates(String s) {
        Stack<Character> st = new Stack<>();
        for(char c:s.toCharArray()){
            if(!st.isEmpty() && st.peek()==c){
                st.pop();
            }
            else{
                st.push(c);
            }
        }
        String ans="";
        while(!st.isEmpty()){
            ans=st.pop()+ans;
        }
        return ans;
    }
}

File: November/Day10/1047._Remove_All_Adjacent_Duplicates_In_String.cpp
class Solution {
public:
    string removeDuplicates(string s) {
      string ans;
      for(int i=0;i<s.size();i++)
      {
         if(ans.back() == s[i])
         {
             ans.pop_back();
          }
          else
          {
              ans+=s[i];
          }
      }
        return ans;
    }
};